core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServiceMethod.java

```diff
@@ -561,6 +563,9 @@ private static String httpHeaderValue(Parameter entry, HttpRequest req) {
         if (clazz == Double.TYPE || clazz == Double.class) {
             return Double.TYPE;
         }
+        if (clazz.isEnum()) {
+            return clazz.asSubclass(Enum.class);
```

Comment:
Do we need `asSubclass()` here? An enum class is always an `Enum` so there seems no need for it.